From: Mel Gorman <mgorman@suse.de>
Date: Thu, 16 Jan 2014 09:48:57 +0000
Subject: [PATCH] mm: numa: avoid unnecessary disruption of NUMA hinting during
 migration pmd-locking

References: Automatic NUMA Balancing (fate#315482)
Patch-mainline: Never, required only for pmd split lock backport

The upstream version of the patch "mm: numa: avoid unnecessary disruption
of NUMA hinting during migration" is not aware of the PMD split locking
which is back-ported to SLE12. This patch reconciles the difference.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 mm/huge_memory.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/huge_memory.c b/mm/huge_memory.c
index 3da5596..f3b25d8 100644
--- a/mm/huge_memory.c
+++ b/mm/huge_memory.c
@@ -1305,7 +1305,7 @@ int do_huge_pmd_numa_page(struct mm_struct *mm, struct vm_area_struct *vma,
 	 * check_same as the page may no longer be mapped.
 	 */
 	if (unlikely(pmd_trans_migrating(*pmdp))) {
-		spin_unlock(&mm->page_table_lock);
+		spin_unlock(ptl);
 		wait_migrate_huge_page(vma->anon_vma, pmdp);
 		goto out;
 	}
